[2024-10-29T14:01:54.232+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-10-29T14:01:54.247+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: pipeline_webscraping.pichau_tasks.Spark_Transformation_pichau manual__2024-10-29T13:59:13.962090+00:00 [queued]>
[2024-10-29T14:01:54.255+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: pipeline_webscraping.pichau_tasks.Spark_Transformation_pichau manual__2024-10-29T13:59:13.962090+00:00 [queued]>
[2024-10-29T14:01:54.256+0000] {taskinstance.py:2865} INFO - Starting attempt 1 of 1
[2024-10-29T14:01:54.271+0000] {taskinstance.py:2888} INFO - Executing <Task(SparkSubmitOperator): pichau_tasks.Spark_Transformation_pichau> on 2024-10-29 13:59:13.962090+00:00
[2024-10-29T14:01:54.279+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=205) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-10-29T14:01:54.280+0000] {standard_task_runner.py:72} INFO - Started process 207 to run task
[2024-10-29T14:01:54.281+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'pipeline_webscraping', 'pichau_tasks.Spark_Transformation_pichau', 'manual__2024-10-29T13:59:13.962090+00:00', '--job-id', '64', '--raw', '--subdir', 'DAGS_FOLDER/DagWebScraping.py', '--cfg-path', '/tmp/tmpu5q5p9gy']
[2024-10-29T14:01:54.283+0000] {standard_task_runner.py:105} INFO - Job 64: Subtask pichau_tasks.Spark_Transformation_pichau
[2024-10-29T14:01:54.327+0000] {task_command.py:467} INFO - Running <TaskInstance: pipeline_webscraping.pichau_tasks.Spark_Transformation_pichau manual__2024-10-29T13:59:13.962090+00:00 [running]> on host 334deecd2258
[2024-10-29T14:01:54.402+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='pipeline_webscraping' AIRFLOW_CTX_TASK_ID='pichau_tasks.Spark_Transformation_pichau' AIRFLOW_CTX_EXECUTION_DATE='2024-10-29T13:59:13.962090+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-10-29T13:59:13.962090+00:00'
[2024-10-29T14:01:54.403+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-10-29T14:01:54.426+0000] {base.py:84} INFO - Retrieving connection 'spark_default'
[2024-10-29T14:01:54.427+0000] {spark_submit.py:327} ERROR - Failed to decode base64 keytab: Incorrect padding
[2024-10-29T14:01:54.429+0000] {spark_submit.py:304} INFO - Could not load connection string spark_default, defaulting to spark://spark:7077
[2024-10-29T14:01:54.431+0000] {spark_submit.py:473} INFO - Spark-Submit cmd: spark-submit --master spark://spark:7077 --conf fs.s3a.access.key=AVcyU1dIcrfI88EqIa7p --conf fs.s3a.secret.key=****** --jars /opt/***/spark/jars/hadoop-aws.jar,/opt/***/spark/jars/aws-java-sdk.jar --name arrow-spark --deploy-mode client /opt/bitnami/spark/src/scripts_spark/WebScraping_Transformation.py --json_file_path s3a://rawzone/pichau/pichau_2024-10-27/data.json
[2024-10-29T14:01:56.312+0000] {spark_submit.py:634} INFO - Warning: Ignoring non-Spark config property: fs.s3a.access.key
[2024-10-29T14:01:56.313+0000] {spark_submit.py:634} INFO - Warning: Ignoring non-Spark config property: fs.s3a.secret.key
[2024-10-29T14:01:56.801+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:56 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2024-10-29T14:01:57.015+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:57 WARN DependencyUtils: Local jar /opt/***/spark/jars/hadoop-aws.jar does not exist, skipping.
[2024-10-29T14:01:57.017+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:57 WARN DependencyUtils: Local jar /opt/***/spark/jars/aws-java-sdk.jar does not exist, skipping.
[2024-10-29T14:01:58.330+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:58 INFO SparkContext: Running Spark version 3.5.2
[2024-10-29T14:01:58.331+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:58 INFO SparkContext: OS info Linux, 5.15.153.1-microsoft-standard-WSL2, amd64
[2024-10-29T14:01:58.331+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:58 INFO SparkContext: Java version 17.0.13
[2024-10-29T14:01:58.361+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:58 INFO ResourceUtils: ==============================================================
[2024-10-29T14:01:58.361+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:58 INFO ResourceUtils: No custom resources configured for spark.driver.
[2024-10-29T14:01:58.362+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:58 INFO ResourceUtils: ==============================================================
[2024-10-29T14:01:58.362+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:58 INFO SparkContext: Submitted application: MyApp
[2024-10-29T14:01:58.388+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:58 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2024-10-29T14:01:58.396+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:58 INFO ResourceProfile: Limiting resource is cpu
[2024-10-29T14:01:58.397+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:58 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2024-10-29T14:01:58.468+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:58 INFO SecurityManager: Changing view acls to: ***
[2024-10-29T14:01:58.469+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:58 INFO SecurityManager: Changing modify acls to: ***
[2024-10-29T14:01:58.469+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:58 INFO SecurityManager: Changing view acls groups to:
[2024-10-29T14:01:58.469+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:58 INFO SecurityManager: Changing modify acls groups to:
[2024-10-29T14:01:58.470+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:58 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: ***; groups with view permissions: EMPTY; users with modify permissions: ***; groups with modify permissions: EMPTY
[2024-10-29T14:01:58.716+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:58 INFO Utils: Successfully started service 'sparkDriver' on port 44589.
[2024-10-29T14:01:58.747+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:58 INFO SparkEnv: Registering MapOutputTracker
[2024-10-29T14:01:58.782+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:58 INFO SparkEnv: Registering BlockManagerMaster
[2024-10-29T14:01:58.804+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:58 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2024-10-29T14:01:58.804+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:58 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2024-10-29T14:01:58.808+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:58 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2024-10-29T14:01:58.834+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:58 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-43836c08-cb95-41bf-93a0-3da500f4d210
[2024-10-29T14:01:58.846+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:58 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2024-10-29T14:01:58.866+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:58 INFO SparkEnv: Registering OutputCommitCoordinator
[2024-10-29T14:01:59.001+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:59 INFO JettyUtils: Start Jetty 0.0.0.0:4040 for SparkUI
[2024-10-29T14:01:59.073+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:59 INFO Utils: Successfully started service 'SparkUI' on port 4040.
[2024-10-29T14:01:59.106+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:59 ERROR SparkContext: Failed to add file:/opt/***/spark/jars/hadoop-aws.jar to Spark environment
[2024-10-29T14:01:59.107+0000] {spark_submit.py:634} INFO - java.io.FileNotFoundException: Jar /opt/***/spark/jars/hadoop-aws.jar not found
[2024-10-29T14:01:59.108+0000] {spark_submit.py:634} INFO - at org.apache.spark.SparkContext.addLocalJarFile$1(SparkContext.scala:2095)
[2024-10-29T14:01:59.109+0000] {spark_submit.py:634} INFO - at org.apache.spark.SparkContext.addJar(SparkContext.scala:2151)
[2024-10-29T14:01:59.109+0000] {spark_submit.py:634} INFO - at org.apache.spark.SparkContext.$anonfun$new$15(SparkContext.scala:521)
[2024-10-29T14:01:59.110+0000] {spark_submit.py:634} INFO - at org.apache.spark.SparkContext.$anonfun$new$15$adapted(SparkContext.scala:521)
[2024-10-29T14:01:59.110+0000] {spark_submit.py:634} INFO - at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[2024-10-29T14:01:59.111+0000] {spark_submit.py:634} INFO - at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[2024-10-29T14:01:59.111+0000] {spark_submit.py:634} INFO - at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[2024-10-29T14:01:59.112+0000] {spark_submit.py:634} INFO - at org.apache.spark.SparkContext.<init>(SparkContext.scala:521)
[2024-10-29T14:01:59.112+0000] {spark_submit.py:634} INFO - at org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
[2024-10-29T14:01:59.113+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2024-10-29T14:01:59.113+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
[2024-10-29T14:01:59.114+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2024-10-29T14:01:59.114+0000] {spark_submit.py:634} INFO - at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
[2024-10-29T14:01:59.115+0000] {spark_submit.py:634} INFO - at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
[2024-10-29T14:01:59.115+0000] {spark_submit.py:634} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
[2024-10-29T14:01:59.115+0000] {spark_submit.py:634} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2024-10-29T14:01:59.116+0000] {spark_submit.py:634} INFO - at py4j.Gateway.invoke(Gateway.java:238)
[2024-10-29T14:01:59.117+0000] {spark_submit.py:634} INFO - at py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
[2024-10-29T14:01:59.117+0000] {spark_submit.py:634} INFO - at py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
[2024-10-29T14:01:59.117+0000] {spark_submit.py:634} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2024-10-29T14:01:59.118+0000] {spark_submit.py:634} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2024-10-29T14:01:59.118+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-10-29T14:01:59.119+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:59 ERROR SparkContext: Failed to add file:/opt/***/spark/jars/aws-java-sdk.jar to Spark environment
[2024-10-29T14:01:59.120+0000] {spark_submit.py:634} INFO - java.io.FileNotFoundException: Jar /opt/***/spark/jars/aws-java-sdk.jar not found
[2024-10-29T14:01:59.120+0000] {spark_submit.py:634} INFO - at org.apache.spark.SparkContext.addLocalJarFile$1(SparkContext.scala:2095)
[2024-10-29T14:01:59.121+0000] {spark_submit.py:634} INFO - at org.apache.spark.SparkContext.addJar(SparkContext.scala:2151)
[2024-10-29T14:01:59.121+0000] {spark_submit.py:634} INFO - at org.apache.spark.SparkContext.$anonfun$new$15(SparkContext.scala:521)
[2024-10-29T14:01:59.121+0000] {spark_submit.py:634} INFO - at org.apache.spark.SparkContext.$anonfun$new$15$adapted(SparkContext.scala:521)
[2024-10-29T14:01:59.122+0000] {spark_submit.py:634} INFO - at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[2024-10-29T14:01:59.122+0000] {spark_submit.py:634} INFO - at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[2024-10-29T14:01:59.123+0000] {spark_submit.py:634} INFO - at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[2024-10-29T14:01:59.123+0000] {spark_submit.py:634} INFO - at org.apache.spark.SparkContext.<init>(SparkContext.scala:521)
[2024-10-29T14:01:59.124+0000] {spark_submit.py:634} INFO - at org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
[2024-10-29T14:01:59.125+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2024-10-29T14:01:59.126+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
[2024-10-29T14:01:59.126+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2024-10-29T14:01:59.127+0000] {spark_submit.py:634} INFO - at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
[2024-10-29T14:01:59.128+0000] {spark_submit.py:634} INFO - at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
[2024-10-29T14:01:59.128+0000] {spark_submit.py:634} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
[2024-10-29T14:01:59.129+0000] {spark_submit.py:634} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2024-10-29T14:01:59.129+0000] {spark_submit.py:634} INFO - at py4j.Gateway.invoke(Gateway.java:238)
[2024-10-29T14:01:59.130+0000] {spark_submit.py:634} INFO - at py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
[2024-10-29T14:01:59.130+0000] {spark_submit.py:634} INFO - at py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
[2024-10-29T14:01:59.131+0000] {spark_submit.py:634} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2024-10-29T14:01:59.132+0000] {spark_submit.py:634} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2024-10-29T14:01:59.132+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-10-29T14:01:59.188+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:59 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://spark:7077...
[2024-10-29T14:01:59.229+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:59 INFO TransportClientFactory: Successfully created connection to spark/172.18.0.4:7077 after 22 ms (0 ms spent in bootstraps)
[2024-10-29T14:01:59.603+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:59 INFO StandaloneSchedulerBackend: Connected to Spark cluster with app ID app-20241029140159-0000
[2024-10-29T14:01:59.613+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:59 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 40077.
[2024-10-29T14:01:59.613+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:59 INFO NettyBlockTransferService: Server created on 334deecd2258:40077
[2024-10-29T14:01:59.615+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:59 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2024-10-29T14:01:59.622+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:59 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 334deecd2258, 40077, None)
[2024-10-29T14:01:59.625+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:59 INFO BlockManagerMasterEndpoint: Registering block manager 334deecd2258:40077 with 434.4 MiB RAM, BlockManagerId(driver, 334deecd2258, 40077, None)
[2024-10-29T14:01:59.628+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:59 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 334deecd2258, 40077, None)
[2024-10-29T14:01:59.629+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:59 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 334deecd2258, 40077, None)
[2024-10-29T14:01:59.649+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:59 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20241029140159-0000/0 on worker-20241029135019-172.18.0.6-45595 (172.18.0.6:45595) with 6 core(s)
[2024-10-29T14:01:59.651+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:59 INFO StandaloneSchedulerBackend: Granted executor ID app-20241029140159-0000/0 on hostPort 172.18.0.6:45595 with 6 core(s), 1024.0 MiB RAM
[2024-10-29T14:01:59.814+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:59 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20241029140159-0000/0 is now RUNNING
[2024-10-29T14:01:59.868+0000] {spark_submit.py:634} INFO - 24/10/29 14:01:59 INFO StandaloneSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.0
[2024-10-29T14:02:00.074+0000] {spark_submit.py:634} INFO - <pyspark.sql.session.SparkSession object at 0x7fc4fd703d10>
[2024-10-29T14:02:00.081+0000] {spark_submit.py:634} INFO - 24/10/29 14:02:00 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2024-10-29T14:02:00.083+0000] {spark_submit.py:634} INFO - 24/10/29 14:02:00 INFO SharedState: Warehouse path is 'file:/opt/***/spark-warehouse'.
[2024-10-29T14:02:01.421+0000] {spark_submit.py:634} INFO - 24/10/29 14:02:01 WARN FileStreamSink: Assume no metadata directory. Error while looking for metadata directory in the path: s3a://rawzone/pichau/pichau_2024-10-28/data.json.
[2024-10-29T14:02:01.421+0000] {spark_submit.py:634} INFO - java.lang.RuntimeException: java.lang.ClassNotFoundException: Class org.apache.hadoop.fs.s3a.S3AFileSystem not found
[2024-10-29T14:02:01.422+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.conf.Configuration.getClass(Configuration.java:2688)
[2024-10-29T14:02:01.423+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem.getFileSystemClass(FileSystem.java:3431)
[2024-10-29T14:02:01.423+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:3466)
[2024-10-29T14:02:01.423+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem.access$300(FileSystem.java:174)
[2024-10-29T14:02:01.424+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem$Cache.getInternal(FileSystem.java:3574)
[2024-10-29T14:02:01.424+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem$Cache.get(FileSystem.java:3521)
[2024-10-29T14:02:01.425+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:540)
[2024-10-29T14:02:01.426+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.Path.getFileSystem(Path.java:365)
[2024-10-29T14:02:01.426+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.streaming.FileStreamSink$.hasMetadata(FileStreamSink.scala:53)
[2024-10-29T14:02:01.427+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.DataSource.resolveRelation(DataSource.scala:366)
[2024-10-29T14:02:01.427+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.loadV1Source(DataFrameReader.scala:229)
[2024-10-29T14:02:01.428+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.$anonfun$load$2(DataFrameReader.scala:211)
[2024-10-29T14:02:01.428+0000] {spark_submit.py:634} INFO - at scala.Option.getOrElse(Option.scala:189)
[2024-10-29T14:02:01.429+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:211)
[2024-10-29T14:02:01.429+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.json(DataFrameReader.scala:362)
[2024-10-29T14:02:01.430+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2024-10-29T14:02:01.430+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2024-10-29T14:02:01.430+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2024-10-29T14:02:01.431+0000] {spark_submit.py:634} INFO - at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[2024-10-29T14:02:01.431+0000] {spark_submit.py:634} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2024-10-29T14:02:01.431+0000] {spark_submit.py:634} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2024-10-29T14:02:01.432+0000] {spark_submit.py:634} INFO - at py4j.Gateway.invoke(Gateway.java:282)
[2024-10-29T14:02:01.432+0000] {spark_submit.py:634} INFO - at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2024-10-29T14:02:01.433+0000] {spark_submit.py:634} INFO - at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2024-10-29T14:02:01.433+0000] {spark_submit.py:634} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2024-10-29T14:02:01.433+0000] {spark_submit.py:634} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2024-10-29T14:02:01.434+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-10-29T14:02:01.434+0000] {spark_submit.py:634} INFO - Caused by: java.lang.ClassNotFoundException: Class org.apache.hadoop.fs.s3a.S3AFileSystem not found
[2024-10-29T14:02:01.435+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.conf.Configuration.getClassByName(Configuration.java:2592)
[2024-10-29T14:02:01.435+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.conf.Configuration.getClass(Configuration.java:2686)
[2024-10-29T14:02:01.435+0000] {spark_submit.py:634} INFO - ... 26 more
[2024-10-29T14:02:01.485+0000] {spark_submit.py:634} INFO - Erro ao ler o arquivo JSON: An error occurred while calling o32.json.
[2024-10-29T14:02:01.486+0000] {spark_submit.py:634} INFO - : java.lang.RuntimeException: java.lang.ClassNotFoundException: Class org.apache.hadoop.fs.s3a.S3AFileSystem not found
[2024-10-29T14:02:01.486+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.conf.Configuration.getClass(Configuration.java:2688)
[2024-10-29T14:02:01.487+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem.getFileSystemClass(FileSystem.java:3431)
[2024-10-29T14:02:01.487+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:3466)
[2024-10-29T14:02:01.488+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem.access$300(FileSystem.java:174)
[2024-10-29T14:02:01.488+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem$Cache.getInternal(FileSystem.java:3574)
[2024-10-29T14:02:01.488+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem$Cache.get(FileSystem.java:3521)
[2024-10-29T14:02:01.489+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:540)
[2024-10-29T14:02:01.489+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.Path.getFileSystem(Path.java:365)
[2024-10-29T14:02:01.490+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.DataSource$.$anonfun$checkAndGlobPathIfNecessary$1(DataSource.scala:724)
[2024-10-29T14:02:01.490+0000] {spark_submit.py:634} INFO - at scala.collection.immutable.List.map(List.scala:293)
[2024-10-29T14:02:01.491+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.DataSource$.checkAndGlobPathIfNecessary(DataSource.scala:722)
[2024-10-29T14:02:01.491+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.DataSource.checkAndGlobPathIfNecessary(DataSource.scala:551)
[2024-10-29T14:02:01.492+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.DataSource.resolveRelation(DataSource.scala:404)
[2024-10-29T14:02:01.492+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.loadV1Source(DataFrameReader.scala:229)
[2024-10-29T14:02:01.493+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.$anonfun$load$2(DataFrameReader.scala:211)
[2024-10-29T14:02:01.494+0000] {spark_submit.py:634} INFO - at scala.Option.getOrElse(Option.scala:189)
[2024-10-29T14:02:01.494+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:211)
[2024-10-29T14:02:01.495+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.json(DataFrameReader.scala:362)
[2024-10-29T14:02:01.495+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2024-10-29T14:02:01.496+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2024-10-29T14:02:01.497+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2024-10-29T14:02:01.497+0000] {spark_submit.py:634} INFO - at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[2024-10-29T14:02:01.497+0000] {spark_submit.py:634} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2024-10-29T14:02:01.498+0000] {spark_submit.py:634} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2024-10-29T14:02:01.498+0000] {spark_submit.py:634} INFO - at py4j.Gateway.invoke(Gateway.java:282)
[2024-10-29T14:02:01.499+0000] {spark_submit.py:634} INFO - at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2024-10-29T14:02:01.499+0000] {spark_submit.py:634} INFO - at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2024-10-29T14:02:01.500+0000] {spark_submit.py:634} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2024-10-29T14:02:01.500+0000] {spark_submit.py:634} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2024-10-29T14:02:01.500+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-10-29T14:02:01.501+0000] {spark_submit.py:634} INFO - Caused by: java.lang.ClassNotFoundException: Class org.apache.hadoop.fs.s3a.S3AFileSystem not found
[2024-10-29T14:02:01.501+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.conf.Configuration.getClassByName(Configuration.java:2592)
[2024-10-29T14:02:01.502+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.conf.Configuration.getClass(Configuration.java:2686)
[2024-10-29T14:02:01.502+0000] {spark_submit.py:634} INFO - ... 29 more
[2024-10-29T14:02:01.503+0000] {spark_submit.py:634} INFO - 
[2024-10-29T14:02:01.503+0000] {spark_submit.py:634} INFO - Traceback (most recent call last):
[2024-10-29T14:02:01.504+0000] {spark_submit.py:634} INFO - File "/opt/bitnami/spark/src/scripts_spark/WebScraping_Transformation.py", line 218, in <module>
[2024-10-29T14:02:01.504+0000] {spark_submit.py:634} INFO - transformation_data(spark,args.json_file_path)
[2024-10-29T14:02:01.504+0000] {spark_submit.py:634} INFO - File "/opt/bitnami/spark/src/scripts_spark/WebScraping_Transformation.py", line 77, in transformation_data
[2024-10-29T14:02:01.505+0000] {spark_submit.py:634} INFO - df=df.withColumn('value',df['value'].cast(DecimalType(12,2)))
[2024-10-29T14:02:01.505+0000] {spark_submit.py:634} INFO - ^^
[2024-10-29T14:02:01.506+0000] {spark_submit.py:634} INFO - UnboundLocalError: cannot access local variable 'df' where it is not associated with a value
[2024-10-29T14:02:01.554+0000] {spark_submit.py:634} INFO - 24/10/29 14:02:01 INFO SparkContext: Invoking stop() from shutdown hook
[2024-10-29T14:02:01.555+0000] {spark_submit.py:634} INFO - 24/10/29 14:02:01 INFO SparkContext: SparkContext is stopping with exitCode 0.
[2024-10-29T14:02:01.565+0000] {spark_submit.py:634} INFO - 24/10/29 14:02:01 INFO SparkUI: Stopped Spark web UI at http://334deecd2258:4040
[2024-10-29T14:02:01.569+0000] {spark_submit.py:634} INFO - 24/10/29 14:02:01 INFO StandaloneSchedulerBackend: Shutting down all executors
[2024-10-29T14:02:01.573+0000] {spark_submit.py:634} INFO - 24/10/29 14:02:01 INFO StandaloneSchedulerBackend$StandaloneDriverEndpoint: Asking each executor to shut down
[2024-10-29T14:02:01.592+0000] {spark_submit.py:634} INFO - 24/10/29 14:02:01 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2024-10-29T14:02:01.617+0000] {spark_submit.py:634} INFO - 24/10/29 14:02:01 INFO MemoryStore: MemoryStore cleared
[2024-10-29T14:02:01.618+0000] {spark_submit.py:634} INFO - 24/10/29 14:02:01 INFO BlockManager: BlockManager stopped
[2024-10-29T14:02:01.628+0000] {spark_submit.py:634} INFO - 24/10/29 14:02:01 INFO BlockManagerMaster: BlockManagerMaster stopped
[2024-10-29T14:02:01.634+0000] {spark_submit.py:634} INFO - 24/10/29 14:02:01 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2024-10-29T14:02:01.650+0000] {spark_submit.py:634} INFO - 24/10/29 14:02:01 INFO SparkContext: Successfully stopped SparkContext
[2024-10-29T14:02:01.650+0000] {spark_submit.py:634} INFO - 24/10/29 14:02:01 INFO ShutdownHookManager: Shutdown hook called
[2024-10-29T14:02:01.651+0000] {spark_submit.py:634} INFO - 24/10/29 14:02:01 INFO ShutdownHookManager: Deleting directory /tmp/spark-095fbadb-cab0-4fa7-a67a-f4b955dc5333
[2024-10-29T14:02:01.655+0000] {spark_submit.py:634} INFO - 24/10/29 14:02:01 INFO ShutdownHookManager: Deleting directory /tmp/spark-095fbadb-cab0-4fa7-a67a-f4b955dc5333/pyspark-68c42207-b4d1-4d13-bb06-02b49e8e646c
[2024-10-29T14:02:01.659+0000] {spark_submit.py:634} INFO - 24/10/29 14:02:01 INFO ShutdownHookManager: Deleting directory /tmp/spark-1c16a534-5ec0-4ee4-b89d-d663df99b31f
[2024-10-29T14:02:01.699+0000] {taskinstance.py:3310} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 560, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark:7077 --conf fs.s3a.access.key=AVcyU1dIcrfI88EqIa7p --conf fs.s3a.secret.key=****** --jars /opt/***/spark/jars/hadoop-aws.jar,/opt/***/spark/jars/aws-java-sdk.jar --name arrow-spark --deploy-mode client /opt/bitnami/spark/src/scripts_spark/WebScraping_Transformation.py --json_file_path s3a://rawzone/pichau/pichau_2024-10-27/data.json. Error code is: 1.
[2024-10-29T14:02:01.705+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=pipeline_webscraping, task_id=pichau_tasks.Spark_Transformation_pichau, run_id=manual__2024-10-29T13:59:13.962090+00:00, execution_date=20241029T135913, start_date=20241029T140154, end_date=20241029T140201
[2024-10-29T14:02:01.717+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-10-29T14:02:01.718+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 64 for task pichau_tasks.Spark_Transformation_pichau (Cannot execute: spark-submit --master spark://spark:7077 --conf fs.s3a.access.key=AVcyU1dIcrfI88EqIa7p --conf fs.s3a.secret.key=****** --jars /opt/***/spark/jars/hadoop-aws.jar,/opt/***/spark/jars/aws-java-sdk.jar --name arrow-spark --deploy-mode client /opt/bitnami/spark/src/scripts_spark/WebScraping_Transformation.py --json_file_path s3a://rawzone/pichau/pichau_2024-10-27/data.json. Error code is: 1.; 207)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3004, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3158, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3182, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 560, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark:7077 --conf fs.s3a.access.key=AVcyU1dIcrfI88EqIa7p --conf fs.s3a.secret.key=****** --jars /opt/***/spark/jars/hadoop-aws.jar,/opt/***/spark/jars/aws-java-sdk.jar --name arrow-spark --deploy-mode client /opt/bitnami/spark/src/scripts_spark/WebScraping_Transformation.py --json_file_path s3a://rawzone/pichau/pichau_2024-10-27/data.json. Error code is: 1.
[2024-10-29T14:02:01.755+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-10-29T14:02:01.807+0000] {taskinstance.py:3900} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-10-29T14:02:01.813+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
